### PR TITLE
Make RayExecutor use the current placement group if one exists

### PR DESCRIPTION
**Pull request by Yard1**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs?
- [x] Did you write any tests to validate this change?  
- [x] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Adds a `PGStrategy` to `RayExecutor`, which will automatically capture the placement group should one be currently present, and use it for Horovod.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

File: horovod/ray/strategy.py [hunk 1]
```diff
@@ -202,3 +203,73 @@ def create_workers(self):
...
+    """
+
+    def __init__(self, *, settings, num_workers, use_gpu, cpus_per_worker,
+                 gpus_per_worker, placement_group=None):
+        self.settings = settings
+        self._num_workers = num_workers
+        self.cpus_per_worker = cpus_per_worker
+        self.gpus_per_worker = gpus_per_worker or 1
+        self.use_gpu = use_gpu
+        self.placement_group = placement_group or get_current_placement_group()
```
[the remark this answers is not in the document]
Comment (richardliaw) [reply horovod/ray/strategy.py:225]: I was thinking maybe we can do:

```
_create_strategy(self):
  if strategy == "Colocated":
    return ColocatedStrategy(**kwargs)
  else:
    PackStrategy(use_current_placement_group=self.use_current_placement_group, **kwargs)

[the remark this answers is not in the document]
Comment (Yard1) [reply horovod/ray/strategy.py]: Done!

File: horovod/ray/strategy.py [hunk 2]
```diff
@@ -135,16 +136,50 @@ def create_workers(self):
         return self.workers, self.get_node_workers(self.workers)
 
 
+def _setup_worker_gpus(workers):
```
Comment (richardliaw) [inline horovod/ray/strategy.py:139]: is this still called?

File: horovod/ray/runner.py [hunk 3]
```diff
@@ -10,7 +11,7 @@
 from horovod.runner.common.util import secret, timeout, hosts
 from horovod.runner.http.http_server import RendezvousServer
 from horovod.ray.utils import detect_nics, nics_to_env_var, map_blocking
-from horovod.ray.strategy import ColocatedStrategy, PackStrategy
+from horovod.ray.strategy import ColocatedStrategy, PackStrategy, PGStrategy
```
Comment (richardliaw) [inline horovod/ray/runner.py:14]: maybe just rename PackSTrategy to PGStrategy?

**Review by richardliaw — COMMENTED** [review]
A couple nits; overall looks good. Ping me when tests pass.

**Comment by Yard1** [comment]
@richardliaw fixed the test, can you rerun?

**Review by richardliaw — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -14,6 +14,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Changed
 
+- By default, RayExecutor will now use the current placement group instead of always creating a new one.
```
Comment (richardliaw) [inline CHANGELOG.md:17]: You'll have to link to the PR I think

**Pull request merged**